### DISCUSSION TITLE
[Proposal] Implement `buildTarget/dependencyResolvers`

### DISCUSSION THREAD

**Discussion by Fristi** [build-server-protocol/build-server-protocol#500] [Proposals]
## Context

For scala-steward we are exploring the possibility to talk BSP.  Scala-steward needs the resolvers to discover which dependencies need to be updated, but this is not supported in the protocol yet.

## Benefits

This allows clients to do analyses on the dependency graph. This allows tools like scala-steward to update dependencies and the project from scalacenter to publish the dependencies to Github or other services which might index for potential vulernabilities

See
- https://github.com/scala-steward-org/scala-steward/issues/2640
- https://github.com/scalacenter/sbt-dependency-graph-action/issues/37

**Comment by adpi2** [comment]
Thanks for the proposal!

Many (or all) build tools do dependency resolution and BSP is about build tools in general. So a `buildTarget/dependencyResolvers` seems to be a legitimate addition to BSP.

 Do you agree @jastice?

**Comment by jastice** [comment]
In principle I agree, but I'm not clear on the specifics - wouldn't a dependency resolver look different depending on the language ecosystem and even particular tool (e.g. maven vs ivy)?
